src/mod_privacy_rdbms.erl

```diff
@@ -311,18 +314,47 @@ execute_privacy_list_insert(LServer, LUser, Name) ->
 execute_delete_data_by_id(LServer, ID) ->
     mongoose_rdbms:execute_successfully(LServer, delete_data_by_id, [ID]).
 
-replace_data_rows(LServer, ID, Rows) when is_integer(ID)->
-    execute_delete_data_by_id(LServer, ID),
-    [mongoose_rdbms:execute_successfully(LServer, privacy_data_insert, [ID|Args])
-     || Args <- Rows],
+replace_data_rows(LServer, ID, []) when is_integer(ID) ->
+    %% Just remove the data, nothing should be inserted
+    execute_delete_data_by_id(LServer, ID);
+replace_data_rows(LServer, ID, Rows) when is_integer(ID) ->
+    {selected, OldRows} = execute_privacy_data_get_by_id(LServer, ID),
+    New = lists:sort(Rows),
+    Old = lists:sort([tuple_to_list(Row) || Row <- OldRows]),
+    Diff = diff_rows(ID, New, Old, []),
+    [mongoose_rdbms:execute_successfully(LServer, Q, Args) || {Q, Args} <- Diff],
     ok.
 
+%% We assume that there are no record duplicates with the same Order.
+%% It's checked in the main module for the New argument.
+%% It's checked by the database for the Old argument.
+diff_rows(ID, [H|New], [H|Old], Ops) ->
+    diff_rows(ID, New, Old, Ops); %% Not modified
+diff_rows(ID, [NH|NT] = New, [OH|OT] = Old, Ops) ->
```

Comment:
Could you extend the variable names as they are quite cryptic? `NewH` etc would suffice.